src/notification_manager/mod.rs

```diff
@@ -490,6 +517,12 @@ impl NotificationManager {
         {
             return Ok(false);
         }
+        if !notification_preferences.hellthread_notifications_enabled
+            && self.is_hellthread_eligible(event.kind())
+            && self.pubkeys_referenced_by_event(event).iter().count() >= MIN_HELLTHREAD_PUBKEYS
```

Comment:
You can just use .len() to count elements in a hashset instead of iterating over every element

Reply:
Fixed.